bin/crew

```diff
@@ -302,40 +330,71 @@ def generate_compatible
   puts 'Generating compatible packages done.'.orange if @opt_verbose
 end
 
-def search (pkgName, silent = false)
-  pkgPath = CREW_PACKAGES_PATH + pkgName + '.rb'
-  begin
-    return set_package(pkgName, pkgPath) if File.exist?(pkgPath)
-  rescue => e
-    puts "Error with #{pkgName}.rb: #{e}".lightred unless e.to_s.include?('uninitialized constant')
-  end
-  unless File.exist?(pkgPath) && silent
+def search (name, silent = false)
+  # search: Same as set_package(), but package name and component name need to be passed in a string
+  #
+  #   name: "<pkgName>::<componentName(optional)>"
+  #
+  pkgName, componentName = name.split('::', 2)
+  componentName ||= 'all'
+  pkgPath = "#{CREW_PACKAGES_PATH}/#{pkgName}.rb"
+
+  unless File.exist?(pkgPath)
     @pkg = nil
-    abort "Package #{pkgName} not found. 😞".lightred unless silent
-  return
+
+    unless silent
+      abort "Package #{pkgName} not found. 😞".lightred
+    else
+      return false
+    end
+  end
+  
+  pkgPath = CREW_PACKAGES_PATH + pkgName + '.rb'
+  set_package(pkgName, componentName)
+
+  unless @pkg.componentFileFilter.has_key?(componentName)
+    unless silent
+      abort "Component #{componentName} not found :/".lightred
+    else
+      return false
+    end
   end
 end
 
-def regexp_search(pkgPat)
-  re = Regexp.new(pkgPat, true)
-  results = Dir[CREW_PACKAGES_PATH + '*.rb'].sort \
-    .select  { |f| File.basename(f, '.rb') =~ re } \
-    .each    { |f| print_package(f, @opt_verbose) }
-  if results.empty?
-    Dir[CREW_PACKAGES_PATH + '*.rb'].each do |packagePath|
-      packageName = File.basename packagePath, '.rb'
-      begin
-        set_package packageName, packagePath
-      rescue => e
-        puts "Error with #{pkgName}.rb: #{e}".red unless e.to_s.include?('uninitialized constant')
+def regexp_search(pattern)
+  regex = Regexp.new(pattern, Regexp::IGNORECASE)
+  any_matches = nil
+
+  Dir.glob("#{CREW_PACKAGES_PATH}/*.rb") do |pkgPath|
```

Comment:
Use `Dir[...]` instead of `Dir.glob(...)`?

Reply:
This should work.  There is no difference.

Reply:
Try `Dir["..."].each`.  We should be consistent with the way we use `Dir` throughout.